templates/d1-openapi/src/index.ts

```diff
@@ -1,178 +1,172 @@
+          name: name,
+          email: email,
         })
         .returning();
 
       return c.json(newUser, 201);
     },
   )
   .get(
-    "/:id",
+    "/users/:id",
     describeRoute({
       responses: {
         200: {
-          content: { "application/json": { schema: resolver(UserSchema) } },
-          description: "User fetched successfully",
+          description: "User queried by ID successfully",
+          content: {
+            "application/json": {
+              schema: resolver(ZUserSelect),
+            },
+          },
         },
       },
     }),
-    zValidator(
-      "param",
-      z.object({
-        id: z.string().uuid().openapi({
-          example: "123e4567-e89b-12d3-a456-426614174000",
-        }),
-      }),
-    ),
+    zodValidator("param", ZUserByIDParams),
     async (c) => {
-      const db = c.get("db");
+      const db = c.var.db;
       const { id } = c.req.valid("param");
+
       const [user] = await db
         .select()
         .from(schema.users)
         .where(eq(schema.users.id, id));
+
       return c.json(user);
     },
   );
 
-// Route Implementations
-app
-  .get(
-    "/",
-    describeRoute({
-      responses: {
-        200: {
-          content: { "text/plain": { schema: resolver(z.string()) } },
-          description: "Root fetched successfully",
-        },
+const app = new Hono()
+  .get("/", (c) => {
+    return c.text("Honc from above! ☁️🪿");
+  })
+  .route("/api", api);
+
+app.onError((error, c) => {
+  console.error(error);
+  if (error instanceof HTTPException) {
+    return c.json(
+      {
+        message: error.message,
       },
-    }),
-    (c) => {
-      return c.text("Honc from above! ☁️🪿");
+      error.status,
+    );
+  }
+
+  return c.json(
+    {
+      message: "Something went wrong",
     },
-  )
-  .route("/api/users", apiRouter);
+    500,
+  );
+});
 
-// Generate OpenAPI spec at /openapi.json
-app
-  .get(
-    "/openapi.json",
-    openAPISpecs(app, {
-      documentation: {
-        info: {
-          title: "D1 Honc! 🪿☁️",
-          version: "1.0.0",
-          description: "D1 Honc! 🪿☁️",
-        },
+/**
+ * Serve a simplified api specification for your API
+ * As of writing, this is just the list of routes and their methods.
+ */
+app.get(
+  "/openapi.json",
+  openAPISpecs(app, {
+    documentation: {
+      info: {
+        title: "HONC D1 App",
+        version: "1.0.0",
       },
-    }),
-  )
-  .use(
-    "/fp/*",
-    createFiberplane({
-      app,
-      openapi: { url: "/openapi.json" },
-    }),
-  );
+    },
+  }),
+);
+
+/**
+ * Mount the Fiberplane api explorer to be able to make requests against your API.
+ *
+ * Visit the explorer at `/fp`
+ */
+app.use(
+  "/fp/*",
+  createFiberplane({
+    app,
+    openapi: { url: "/openapi.json" },
+  }),
+);
 
 export default app;
+
+// Export the instrumented app if you've wired up a Fiberplane-Hono-OpenTelemetry trace collector
+//
+// export default instrument(app);
```

Comment:
```suggestion
```

we can remove instrument for now - setting up a tracing sidecar is a bit of a pain right now, and only reliable locally